lib/logstash/inputs/http.rb

```diff
@@ -75,11 +83,20 @@ def register
   end # def register
 
   def run(queue)
```

Comment:
:+1: on logging unauthorized access. also +1 on many other things we can add.
Even with basic auth it would be nice to add different user/password to different "realms" or fractions of the url namespace.

This could be done with an hash
```
{ 
  regexp1 => [user1, pass1],
  regexp2 => [user2, pass2],
  ...
}
```